File: Python/0724_FindPivotIndex.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left = 0
        right = sum(nums)

        for i in range(0, len(nums)):
            right -= nums[i]
            if left == right: 
                return i
            left += nums[i]
        
        right -= nums[len(nums)-1]
        if left == right: 
            return len(nums)-1
        else: 
            return -1
        

File: Csharp/0724_FindPivotIndex.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        int left = 0;
        int right = nums.Sum();

        for (int i = 0; i < nums.Length; i++)
        {
            right -= nums[i];
            if (left == right) return i;
            left += nums[i];
        }
        
        right -= nums[nums.Length-1];
        if (left == right) return nums.Length-1;
        else return -1;
    }
}